pkg/apiserver/internal/v1/util.go

```diff
@@ -158,3 +165,35 @@ func convertListMeta(in metav1.ListMeta) (out *v1.ListMeta) {
 	}
 	return
 }
+
+type ConvertFunc func(runtime.Object) (*any.Any, error)
+type SendFunc func(*v1.WatchEvent) error
+
+func watch(ctx context.Context, client dynamic.Interface, gvr schema.GroupVersionResource, namespace string, opts metav1.ListOptions, sendFunc SendFunc, convertFunc ConvertFunc) error {
```

Comment:
why do we need to pass `ctx` and `sendFunc` to the function? Why don't we just pass `stream` to the function?

Reply:
because each server has own stream, with a different type
https://github.com/kubermatic/kubecarrier/blob/ace10a9bd4cfd2287a523dc9b588371c50270574/pkg/apiserver/internal/v1/offering.go#L125.
we can add create an interface with `Context` and `Send` methods and pass `stream` as it implements it, but I think it will add complexity.

Reply:
For watch function working properly, the context should always be the `Stream.Context()` right? If someone pass a random context to it, it will not work I guess, so I think it is better to pass the stream to it? WDYT?

Reply:
I made this change, pls take a look.
I also like it, fewer params to pass